### PR TITLE
Add missing config params validation

### LINKED ISSUE #8465: Add additional validation to config parameters
### Description

Add additional range validation for the genesis config and module configs as below

#### Genesis

| Property | Min | Max |
|----------|-----|-----|
|      bftBatchSize    | n/a    |   MAX_NUM_VALIDATORS+2  |
|       blockTime   |  3 (tentative value)   |   `30*24*60*60` 1 block per month  |
|       maxTransactionsSize   |  MAX_CCM_SIZE + some buffer   |   30*1024 (should be benchmarked)  |

#### PoS
| Property | Min | Max |
|----------|-----|-----|
|      numberActiveValidators    | 1    |   MAX_NUM_VALIDATORS  |
|       numberStandbyValidators   |  0   |   2  |
|       factorSelfStakes   |  1   |   MAX_UINT32  |
|      maxLengthName    | 10   | 30   |
|     maxNumberSentStakes     |  1  |  20  |
|    maxNumberPendingUnlocks      |  1  |  40  |
|     failSafeMissedBlocks     |   2 | MAX_UINT64   |
|      failSafeInactiveWindow    | 432000/BLOCK_TIME   | 31536000/BLOCK_TIME   |
|     punishmentWindow     |  432000/BLOCK_TIME   | 31536000/BLOCK_TIME  |
|     minWeightStandby     |  1   | MAX_UINT64 |
|     validatorRegistrationFee     |  0   | MAX_UINT64 |
|     maxBFTWeightCap     |  300   | 10000 |
|     commissionIncreasePeriod     |  100   | 10000 |
|     maxCommissionIncreaseRate     |  0   | MAX_UINT64 |

##### Additional properties which are currently not configurable

- PUNISHMENT_WINDOW_STAKING
- PUNISHMENT_WINDOW_SELF_STAKING
- LOCKING_PERIOD_STAKING
- LOCKING_PERIOD_SELF_STAKING
- BASE_STAKE_AMOUNT
- WEIGHT_SCALE_FACTOR
- REPORT_MISBEHAVIOR_REWARD
- REPORT_MISBEHAVIOR_LIMIT_BANNED

### Acceptance Criteria
- Validation for above configuration are added along with test

### PR DESCRIPTION
**Pull request by bobanm**
### What was the problem?

This PR resolves #8465 but with [additional comments](https://github.com/LiskHQ/lisk-sdk/issues/8465#issuecomment-1623661324). Please read before reviewing the PR.

### How was it solved?

Added validation as per the requirement and per additional comment in the ticket.

### CODE REVIEW

File: framework/src/modules/pos/schemas.ts [hunk 1]
```diff
@@ -133,22 +134,30 @@ export const configSchema = {
...
 		},
 		maxNumberPendingUnlocks: {
 			type: 'integer',
 			format: 'uint32',
+			minimum: 1,
+			maximum: 40,
 		},
 		failSafeMissedBlocks: {
 			type: 'integer',
 			format: 'uint32',
```
Comment (Incede) [inline framework/src/modules/pos/schemas.ts:159]: I believe this should be `uint64`

**Review by Incede — CHANGES_REQUESTED** [review]

File: framework/src/schema/application_config_schema.ts [hunk 2]
```diff
@@ -235,12 +241,13 @@ export const applicationConfigSchema = {
 				},
 				blockTime: {
 					type: 'number',
-					minimum: 2,
+					minimum: 3,
```
[the remark this answers is not in the document]
Comment (bobanm) [reply framework/src/schema/application_config_schema.ts:244]: The comment means that we shouldn't use block time as a value to calculate another parameter's min/max range. For example, setting min for `failSafeInactiveWindow` to `432000 / BLOCK_TIME` would not be valid.

But it is still required to validate that block time provided by user config is within allowed ranges.

File: framework/src/constants.ts [hunk 3]
```diff
@@ -37,3 +37,6 @@ export const MAX_BLOCK_CACHE = 515;
 export const DEFAULT_MAX_INBOUND_CONNECTIONS = 100;
 export const DEFAULT_MAX_OUTBOUND_CONNECTIONS = 20;
 export const OWNER_READ_WRITE = 0o600;
+
+export const MAX_CCM_SIZE = 10240;
```
[the remark this answers is not in the document]
Comment (bobanm) [reply framework/src/constants.ts:41]: The problem is that in this case the constant is defined in one module [interop] and needed in another one [PoS] or on the application level.

There is a quest #8712 to move all constants that are needed by multiple modules to framework level. We will tackle that after bug fixing. Until then, we agreed with @ishantiw to duplicate framework-level constants to framework, so we don't introduce potential circular dependencies.

File: framework/src/schema/application_config_schema.ts [hunk 4]
```diff
@@ -250,9 +257,7 @@ export const applicationConfigSchema = {
 				},
 				maxTransactionsSize: {
 					type: 'integer',
-					// eslint-disable-next-line @typescript-eslint/no-magic-numbers
-					minimum: 10 * 1024, // Kilo Bytes
-					// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+					minimum: MAX_CCM_SIZE + 1024,
 					maximum: 30 * 1024, // Kilo Bytes
```
[the remark this answers is not in the document]
Comment (bobanm) [reply framework/src/schema/application_config_schema.ts:261]: I agreed with @ricott1 to use 1kb as the buffer, and if a bigger buffer is needed, we will solve it in a separate quest.

File: framework/src/modules/pos/schemas.ts (hunk 1 above)
Comment (bobanm) [reply framework/src/modules/pos/schemas.ts:159]: @gkoumout I see that in the param spreadsheet you defined `FAIL_SAFE_MISSED_BLOCKS` to have max value `MAX_UINT_64`.

While [LIP-0057](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#notation-and-constants) defines it as `uint32`.

Which one should be corrected?

Comment (gkoumout) [reply framework/src/modules/pos/schemas.ts:159]: I guess uint32 is the right thing. We use uint64 only for constants that correspond to tokens, e.g., fee, validator weight, etc, the rest is all uint32 I think.

**Review by Incede — APPROVED** [review]

**Review by mosmartin — APPROVED** [review]

**Pull request merged**